crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -141,44 +130,52 @@ impl ModuleResolveResult {
                 .into_iter()
                 .map(ModuleResolveResultItem::OutputAsset)
                 .collect(),
-            references: Vec::new(),
+            affecting_sources: Vec::new(),
         }
     }
 
-    pub fn modules_with_references(
+    pub fn modules_with_affecting_sources(
         modules: Vec<Vc<Box<dyn Module>>>,
-        references: Vec<Vc<Box<dyn ModuleReference>>>,
+        affecting_sources: Vec<Vc<Box<dyn Source>>>,
     ) -> ModuleResolveResult {
         ModuleResolveResult {
             primary: modules
                 .into_iter()
                 .map(ModuleResolveResultItem::Module)
                 .collect(),
-            references,
+            affecting_sources,
         }
     }
 
-    pub fn add_reference_ref(&mut self, reference: Vc<Box<dyn ModuleReference>>) {
-        self.references.push(reference);
+    pub fn primary_modules_ref(&self) -> impl Iterator<Item = Vc<Box<dyn Module>>> + '_ {
```

Comment:
```suggestion
    pub fn primary_modules_iter(&self) -> impl Iterator<Item = Vc<Box<dyn Module>>> + '_ {
```
not a fan of the name here and below, but also don't care that much